tests/src/test/scala/com/nvidia/spark/rapids/CastOpSuite.scala

```diff
@@ -648,38 +630,56 @@ object CastOpSuite {
     )
 
     // invalid values that should be cast to null on both CPU and GPU
-    val invalidValues = Seq(
-      "200", // year too few digits
-      "20000", // year too many digits
-      "1999\rGARBAGE",
-      "1999-1\rGARBAGE",
-      "1999-12\rGARBAGE",
-      "1999-12-31\rGARBAGE",
-      "1999-10-1 TGARBAGE\nMORE GARBAGE",
-      "200-1-1",
-      "2000-1-1-1",
-      "2000-1-1-1-1",
-      "-1-1-1-",
-      "2010-01-6\r\nT\r\n12:34:56.000111Z",
-      "2010-01-6\nT 12:34:56.000111Z",
-      "2010-01-6\nT\n12:34:56.000111Z",
-      "2010-01-6 T 12:34:56.000111Z",
-      "2010-01-6\tT\t12:34:56.000111Z",
-      "2010-01-6T\t12:34:56.000111Z",
-      "2010-01-6T 12:34:56.000111Z",
-      "2010-01-6T  12:34:56.000111Z",
-      "2010-01-6 T 12:34:56.000111Z",
-      "2010-01-6  T12:34:56.000111Z",
-      "2010-01-6 ",
-      "2010-01-6 T",
-      "2010-01-6 T\n",
-      "2010-01-6 T\n12:34:56.000111Z",
-      "2018random_text",
-      "2018-11random_text",
-      "2018-1random_text",
-      "2018-11-08random_text",
-      "2018-11-9random_text"
-    )
+    val invalidValues = if (validOnly) {
+      Seq.empty
+    } else {
+      Seq(
+        "200", // year too few digits
+        "20000", // year too many digits
+        "1999\rGARBAGE",
+        "1999-1\rGARBAGE",
+        "1999-12\rGARBAGE",
+        "1999-12-31\rGARBAGE",
+        "1999-10-1 TGARBAGE\nMORE GARBAGE",
+        "200-1-1",
+        "2000-1-1-1",
+        "2000-1-1-1-1",
+        "-1-1-1-",
+        "2010-01-6\r\nT\r\n12:34:56.000111Z",
+        "2010-01-6\nT 12:34:56.000111Z",
+        "2010-01-6\nT\n12:34:56.000111Z",
+        "2010-01-6 T 12:34:56.000111Z",
+        "2010-01-6\tT\t12:34:56.000111Z",
+        "2010-01-6T\t12:34:56.000111Z",
+        "2010-01-6T 12:34:56.000111Z",
+        "2010-01-6T  12:34:56.000111Z",
+        "2010-01-6 T 12:34:56.000111Z",
+        "2010-01-6  T12:34:56.000111Z",
+        "2010-01-6 ",
+        "2010-01-6 T",
+        "2010-01-6 T\n",
+        "2010-01-6 T\n12:34:56.000111Z",
+        "2018random_text",
+        "2018-11random_text",
+        "2018-1random_text",
+        "2018-11-08random_text",
+        "2018-11-9random_text",
+        // `yyyy-[m]m-[d]dT*` in some version of Spark this is valid and in others it is not
```

Comment:
It looks like the only time we are excluding these values from the test is when `ansiEnabled=true` rather than excluding them for particular Spark versions?

Reply:
Yes, version 3.1 + when ANSI is enabled they now cause an exception to be thrown.  Because the test this is for is "Test all supported casts with in-range values" I assumed that is fine. If you want me to allow different values for different versions I can.

Reply:
I think this change is good but the comment confused me so I just wanted to make sure that I understood this correctly.

Reply:
I'll update the comment to explain what is happening.